src/scripts/main.js

```diff
@@ -1,3 +1,150 @@
 'use strict';
 
-// write code here
+const phonesUrl
+  = 'https://mate-academy.github.io/phone-catalogue-static/api/phones.json';
+const phonesDetailsUrl
+  = 'https://mate-academy.github.io/phone-catalogue-static/api/phones/';
+
+// this function is necessary only for phones ids receiving
+
+function getPhones(url) {
+  return fetch(url)
+    .then(response => {
+      if (!response.ok) {
+        setTimeout(() => {
+          // throw `${response.status} - ${response.statusText}`;
+        }, 5000);
+      }
+
+      return response.json();
+    });
+}
+
+function getFirstReceivedDetails(phonesID) {
+  const promisesArray = [];
+
+  for (const id of phonesID) {
+    promisesArray.push(fetch(`${phonesDetailsUrl}${id}.json`));
+  }
+
+  const firstReceived = Promise.race(promisesArray);
+
+  firstReceived.then(response => response.json())
+    .then(phoneDetails => {
+      makeDom(phoneDetails.id, phoneDetails.name,
+        'First received', 'first-received');
+    })
+    .catch(error => {
+      makeDom('Error:', error);
+    });
+
+  return firstReceived;
+}
+
+function getAllSuccessfulDetails(phonesID) {
+  const phoneDetailsArray = [];
+  const promisesArray = [];
+
+  for (const id of phonesID) {
+    promisesArray.push(fetch(`${phonesDetailsUrl}${id}.json`));
+  }
+
+  Promise.allSettled(promisesArray).then((responses) => {
+    const successfullResponses
+    = responses.filter((response) => response.status === 'fulfilled');
+
+    for (const result of successfullResponses) {
+      result.value.json().then(phoneDetails => {
```

Comment:
Do you need to run `.json()` on Promise value? If I remember everything correctly, you don't need to do this if you use `async/await`. But no strong opinion, maybe I forgot something)

Reply:
I haven't tried to make the task using async/await. Maybe you are right.